lib/Myriad/Bootstrap.pm

```diff
@@ -78,6 +80,55 @@ sub allow_modules {
     @ALLOWED_MODULES{@_} = (1) x @_;
 }
 
+sub open_pipe {
+    # Establish comms channel for child process
+    socketpair my $child_pipe, my $parent_pipe, $constant{AF_UNIX}, $constant{SOCK_STREAM}, $constant{PF_UNSPEC}
+        or die $!;
+
+    { # Unbuffered writes
+        my $old = select($child_pipe);
+        $| = 1; select($parent_pipe);
+        $| = 1; select($old);
+    }
+
+    make_pipe_nonblocking($parent_pipe);
+    make_pipe_nonblocking($child_pipe);
```

Comment:
If you're moving this out, might as well move the unbuffered write setup into functions as well?